File: April-18/27-04-18/python/02-Little-Shino-and-Pairs.py
n = int(input())
ar = [int(x) for x in input().split()]

st = []
ans = 0
st.append(ar[0])

for i in range(1, n):
    
    while(len(st) != 0 and ar[i] > st[-1]):
        st.pop()
        ans += 1
    if(len(st) != 0):
        ans += 1
    st.append(ar[i])
print(ans)

File: April-18/27-04-18/java/02-Little-Shino-and-Pairs.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int ar[] = new int[n];
        for(int i = 0; i < n; i++)
            ar[i] = input.nextInt();
        Stack<Integer> st = new Stack<Integer>();
        st.push(ar[0]);
        int ans = 0;
        for(int i = 1; i < n; i++)
        {
            while(!st.empty() && ar[i] > st.peek())
            {
                st.pop();
                ans++;
            }
            if(!st.empty())
                ans++;
            st.push(ar[i]);
        }
        System.out.println(ans);
        
        
    }
}
